receiver/cloudflarereceiver/logs.go

```diff
@@ -0,0 +1,305 @@
+			host.ReportFatalError(err)
+		}
+	}()
+	return nil
+}
+
+func (l *logsReceiver) handleRequest(rw http.ResponseWriter, req *http.Request) {
+	if l.cfg.Secret != "" {
+		secretHeader := req.Header.Get(secretHeaderName)
+		if secretHeader == "" {
+			rw.WriteHeader(http.StatusUnauthorized)
+			l.logger.Debug("Got payload with no Secret when it was specified in config, dropping...")
+			return
+		} else if secretHeader != l.cfg.Secret {
+			rw.WriteHeader(http.StatusUnauthorized)
+			l.logger.Debug("Got payload with invalid Secret, dropping...")
+			return
+		}
+	}
+
+	var payload []byte
+	if req.Header.Get("Content-Encoding") == "gzip" {
+		reader, err := gzip.NewReader(req.Body)
+		if err != nil {
+			rw.WriteHeader(http.StatusUnprocessableEntity)
+			l.logger.Debug("Got payload with gzip, but failed to read", zap.Error(err))
+			return
+		}
+		defer reader.Close()
+		// Read the decompressed response body
+		payload, err = io.ReadAll(reader)
+		if err != nil {
+			rw.WriteHeader(http.StatusUnprocessableEntity)
+			l.logger.Debug("Got payload with gzip, but failed to read", zap.Error(err))
+			return
+		}
+	} else {
+		var err error
+		payload, err = io.ReadAll(req.Body)
+		if err != nil {
+			rw.WriteHeader(http.StatusUnprocessableEntity)
+			l.logger.Debug("Failed to read alerts payload", zap.Error(err), zap.String("remote", req.RemoteAddr))
+			return
+		}
+	}
+
+	if string(payload) == "test" {
+		l.logger.Info("Received test request from Cloudflare")
+		rw.WriteHeader(http.StatusOK)
+		return
+	}
+
+	logs, err := parsePayload(payload)
+	if err != nil {
+		rw.WriteHeader(http.StatusUnprocessableEntity)
+		l.logger.Error("Failed to convert cloudflare request payload to maps", zap.Error(err))
+		return
+	}
+
+	if err := l.consumer.ConsumeLogs(req.Context(), l.processLogs(pcommon.NewTimestampFromTime(time.Now()), logs)); err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		l.logger.Error("Failed to consumer alert as log", zap.Error(err))
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+}
+
+func parsePayload(payload []byte) ([]map[string]interface{}, error) {
+	var logs []map[string]interface{}
+	for _, line := range bytes.Split(payload, []byte("\n")) {
+		if len(line) == 0 {
+			continue
+		}
+		var log map[string]interface{}
+		err := json.Unmarshal(line, &log)
+		if err != nil {
+			return logs, err
+		}
+		logs = append(logs, log)
+	}
+	return logs, nil
+}
+
+func (l *logsReceiver) processLogs(now pcommon.Timestamp, logs []map[string]interface{}) plog.Logs {
+	pLogs := plog.NewLogs()
+
+	// Group logs by ZoneName field if it was configured so it can be used as a resource attribute
+	groupedLogs := make(map[string][]map[string]interface{})
+	for _, log := range logs {
+		zone := ""
+		if v, ok := log["ZoneName"]; ok {
+			if stringV, ok := v.(string); ok {
+				zone = stringV
+			}
+		}
+		groupedLogs[zone] = append(groupedLogs[zone], log)
+	}
+
+	for zone, logGroup := range groupedLogs {
```

Comment:
Would you mind adding another test case with logs from a couple different `ZoneName`'s, so we can validate that we're seeing multiple resources as expected?